staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -316,28 +359,51 @@ type fsStatusUpdate struct {
+func (cfgCtl *configController) digestConfigObjects(newPLs []*fctypesv1a1.PriorityLevelConfiguration, newFSs []*fctypesv1a1.FlowSchema) (bool, bool, time.Duration, error) {
+	fsStatusUpdates, wait := cfgCtl.lockAndDigestConfigObjects(newPLs, newFSs)
+	var triedWrites, didWrites bool
 	var errs []error
+	fieldManager := "api-priority-and-fairness-config-consumer-v1"
+	if cfgCtl.invertDangle {
+		fieldManager += "x"
+	}
 	for _, fsu := range fsStatusUpdates {
 		enc, err := json.Marshal(fsu.condition)
 		if err != nil {
 			// should never happen because these conditions are created here and well formed
 			panic(fmt.Sprintf("Failed to json.Marshall(%#+v): %s", fsu.condition, err.Error()))
 		}
-		klog.V(4).Infof("Writing Condition %s to FlowSchema %s because its previous value was %s", string(enc), fsu.flowSchema.Name, fcfmt.Fmt(fsu.oldValue))
-		_, err = cfgCtl.flowcontrolClient.FlowSchemas().Patch(context.TODO(), fsu.flowSchema.Name, apitypes.StrategicMergePatchType, []byte(fmt.Sprintf(`{"status": {"conditions": [ %s ] } }`, string(enc))), metav1.PatchOptions{FieldManager: "api-priority-and-fairness-config-consumer-v1"}, "status")
-		if err != nil {
+		klog.V(4).Infof("%s writing Condition %s to FlowSchema %s, which had ResourceVersion=%s, because its previous value was %s", cfgCtl.name, string(enc), fsu.flowSchema.Name, fsu.flowSchema.ResourceVersion, fcfmt.Fmt(fsu.oldValue))
+		triedWrites = true
+		fs2, err := cfgCtl.flowcontrolClient.FlowSchemas().Patch(context.TODO(), fsu.flowSchema.Name, apitypes.StrategicMergePatchType, []byte(fmt.Sprintf(`{"status": {"conditions": [ %s ] } }`, string(enc))), metav1.PatchOptions{FieldManager: fieldManager}, "status")
+		if err == nil {
+			cfgCtl.noteStatusUpdate(fsu.flowSchema, fs2)
+			didWrites = true
+		} else if apierrors.IsNotFound(err) || apierrors.IsConflict(err) {
+			// This object has been deleted or the server has a newer
+			// revision.  In either case a notification is coming and
+			// nothing more needs to be done here.
+			why := "NotFound"
+			if apierrors.IsConflict(err) {
+				why = "Conflict"
+			}
+			klog.V(5).Infof("%s at %s: attempted update of FlowSchema %s from ResourceVersion %s failed because %s", cfgCtl.name, cfgCtl.clock.Now().Format(timeFmt), fsu.flowSchema.Name, fsu.flowSchema.ResourceVersion, why)
+		} else {
 			errs = append(errs, errors.Wrap(err, fmt.Sprintf("failed to set a status.condition for FlowSchema %s", fsu.flowSchema.Name)))
 		}
 	}
 	if len(errs) == 0 {
-		return nil
+		return triedWrites, didWrites, wait, nil
 	}
-	return apierrors.NewAggregate(errs)
+	return triedWrites, didWrites, wait, errorutils.NewAggregate(errs)
+}
+
+func (cfgCtl *configController) noteStatusUpdate(fs, fs2 *fctypesv1a1.FlowSchema) {
+	klog.V(5).Infof("%s at %s: successful update of FlowSchema %s from ResourceVersion %s to %s", cfgCtl.name, cfgCtl.clock.Now().Format(timeFmt), fs.Name, fs.ResourceVersion, fs2.ResourceVersion)
```

Comment:
it's a little weird to output your own time.  Pretty sure klog already gives you time in the ouptut, doesn't it?

Reply:
When testing with a fake clock, the klog framing does not include the time that matters.